File: Ahorn/triggers/MomentumCameraOffsetTrigger.jl
module FurryHelperMomentumCameraOffset

using ..Ahorn, Maple

@mapdef Trigger "FurryHelper/momentumCameraOffsetTrigger" MomentumCameraOffsetTrigger(x::Integer, y::Integer, width::Integer=Maple.defaultTriggerWidth, height::Integer=Maple.defaultTriggerHeight,
    offsetXFrom::Number=0.0, offsetXTo::Number=0.0, offsetYFrom::Number=0.0, offsetYTo::Number=0.0,
    momentumFrom::Number=0.0, momentumTo::Number=0.0,
    momentumMode::String="VerticalMomentum", onlyOnce::Bool=false, xOnly::Bool=false, yOnly::Bool=false)

const placements = Ahorn.PlacementDict(
    "Momentum Camera Offset (FurryHelper)" => Ahorn.EntityPlacement(
        MomentumCameraOffsetTrigger,
        "rectangle",
    ),
)

function Ahorn.editingOptions(trigger::MomentumCameraOffsetTrigger)
    return Dict{String, Any}(
        "momentumMode" => String[
            "HorizontalMomentum",
            "VerticalMomentum"
        ]
    )
end

end

File: Loenn/triggers/momentum_camera_offset_trigger.lua
local momentumCameraOffsetTrigger = {}

momentumCameraOffsetTrigger.name = "FurryHelper/momentumCameraOffsetTrigger"

momentumCameraOffsetTrigger.placements = {
    name = "momentum_camera",
    data = {
        offsetXFrom = 0.0,
        offsetXTo = 0.0,
        offsetYFrom = 0.0,
        offsetYTo = 0.0,
        momentumFrom = 0.0,
        momentumTo = 0.0,
        momentumMode = "VerticalMomentum",
        onlyOnce = false,
        xOnly = false,
        yOnly = false
    }
}

momentumCameraOffsetTrigger.fieldInformation = {
    momentumMode = {
        options = {
            "HorizontalMomentum",
            "VerticalMomentum"
        },
        editable = false
    }
}

momentumCameraOffsetTrigger.fieldOrder = {
    "x", "y", "width", "height", 
    "offsetXFrom", "offsetXTo", "offsetYFrom", "offsetYTo", 
    "momentumFrom", "momentumTo", "momentumMode", 
    "onlyOne", "xOnly", "yOnly"
}

return momentumCameraOffsetTrigger